module/zfs/zil.c

```diff
@@ -3259,29 +3262,23 @@ zil_suspend(const char *osname, void **cookiep)
 	zil_commit_impl(zilog, 0);
 
 	/*
-	 * Now that we've ensured all lwb's are LWB_STATE_DONE, we use
-	 * txg_wait_synced() to ensure the data from the zilog has
-	 * migrated to the main pool before calling zil_destroy().
+	 * Now that we've ensured all lwb's are LWB_STATE_DONE,
+	 * txg_wait_synced() will be called from within zil_destroy(),
+	 * which will ensure the data from the zilog has migrated to the
+	 * main pool before it returns.
 	 */
-	txg_wait_synced(zilog->zl_dmu_pool, 0);
-
 	zil_destroy(zilog, B_FALSE);
 
 	mutex_enter(&zilog->zl_lock);
 	zilog->zl_suspending = B_FALSE;
 	cv_broadcast(&zilog->zl_cv_suspend);
 	mutex_exit(&zilog->zl_lock);
 
+	/*
+	 * Encrypted datasets need to wait for all data to be
+	 * synced out before removing the mapping.
```

Comment:
This comment seems like a non-sequitur, since we are not syncing something out here?  Maybe reference back to the zil_destroy()?  e.g.
`zil_destroy() has already waited for all dirty data to be synced out, so it's safe to remove the mapping now.`

Reply:
Good catch. I had not read the comment when I moved it.